lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart

```diff
@@ -588,6 +588,9 @@ class MailboxDashBoardController extends ReloadableController with UserSettingPo
 
   void setSelectedMailbox(PresentationMailbox? newPresentationMailbox) {
     log('MailboxDashBoardController::setSelectedMailbox: SELECTED_MAILBOX_ID = ${newPresentationMailbox?.id.asString} |  SELECTED_MAILBOX_NAME = ${newPresentationMailbox?.name?.name} | ');
+    if (selectedMailbox.value?.id != newPresentationMailbox?.id) {
+      dispatchEmailUIAction(GetAllEmailAction(newPresentationMailbox));
+    }
```

Comment:
IMO, it is not clean. select function should only select, dont need to do other thing. Get all email should move back to listener of this change.